classifier.py

```diff
@@ -0,0 +1,129 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of ISTEX_MentalRotation.
+# Copyright (C) 2016 3ST ERIC Laboratory.
+#
+# This is a free software; you can redistribute it and/or modify it
+# under the terms of the Revised BSD License; see License file for
+# more details.
+
+# Binary classifier for Mental_Rotation articles
+
+# co-author: Hussein AL-NATSHEH <hussein.al-natsheh@ish-lyon.cnrs.fr>
+# Affiliation:{University of Lyon, ERIC Laboratory, CNRS, ISH-Lyon}, France
+
+# co-author : Lucie Martinet <lucie.martinet@univ-lorraine.fr>
+# Affiliation: ERIC Laboratory, France
+# Thanks to ISTEX project for the funding
+
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.model_selection import train_test_split
+from sklearn.metrics import accuracy_score
+import numpy as np
+import random, pickle, argparse, json, os
+
+
+def sampling(lst, class_type, sample_size=500):
+	# given a list, return a sample list of a given size with a class label for
+	# each sample
+	lst_size = len(lst)
+	if sample_size > lst_size:
+		indecies = np.random.choice(range(lst_size),size=sample_size - lst_size)
+		sample_lst = np.vstack((lst,lst[indecies]))
+	else:
+		indecies = random.sample(range(lst_size), sample_size)
+		sample_lst = lst[indecies]
+
+	if class_type == 'positive':
+		labels = np.ones(sample_size, dtype=np.int)
+	elif class_type == 'negative':
+		labels = np.zeros(sample_size, dtype=np.int)
+	else:
+		raise NameError('Unrecognized class type. \
+		       Must be either "positive" or "negative"')
+	labels = labels.reshape(-1, 1)
+	sample = np.hstack((sample_lst, labels))
+	return sample
+
+def balanced_dataset(pos_lst, neg_lst, size):
+	pos_size = size/2
+	neg_size = size - pos_size # in case of odd int value of size
+	pisitives = sampling(pos_lst, 'positive', pos_size)
+	negatives = sampling(neg_lst, 'negative', neg_size)
+	return np.vstack((pisitives,negatives))
+
+
+def search_top_relevant(sim_results, inverse_ndx, sim_th):
+	related = sim_results[:,1] > sim_th
+	related_ndx = []
+	for i, r in enumerate(related):
+		if r:
+			related_ndx.append(i)
+	results = []
+	for i in related_ndx:
+	    results.append(inverse_ndx[str(count_pos + i)])
+	return np.unique(results)
+
+if __name__ == "__main__" :
+	parser = argparse.ArgumentParser()
+	# the file contains the documents decomposed vectors (model) to load
+	parser.add_argument("--vectors", default="results/output_svd.pickle", type=str)
+	parser.add_argument("--inv_ndx", default="results/output_paragraph_inverse_index.json", type=str)
+	parser.add_argument("--count_pos", default=183, type=int)
```

Comment:
better use cnt_positive instead of count_pos or add comments to explain what it is.